daemon/cluster/cluster.go

```diff
@@ -345,11 +344,19 @@ func (c *Cluster) Cleanup() {
 			}
 		}
 	}
+
+	// Release the lock before stopping the node, so that calls back into
+	// Cluster during the shutdown process do not hang.
+	c.mu.Unlock()
```

Comment:
"currentNodeState should not be called without a read lock"

But I moved the unlock just below that.